R/util.R

```diff
@@ -513,3 +513,34 @@ package_root <- function(path) {
       length(grep('^Package: ', readLines(desc))) == 0) return(package_root(dir))
   dir
 }
+
+
+# retrieve package version without fear of error
+# loading namespace is ok as these packages have been or will be used
+get_package_version_string <- function(package) {
+  tryCatch(
+    as.character(getNamespaceVersion(package)),
+    error = function(e) {
+      NULL
+    }
+  )
+}
+# find all loaded packages.
+# May contain extra packages, but contains all packages used while knitting
+get_loaded_packages <- function() {
+  base_r_packages <- c(
+    "base", "compiler", "datasets", "graphics", "grDevices",
+    "grid", "methods", "parallel", "splines",
+    "stats", "stats4", "tcltk", "tools", "utils"
+  )
```

Comment:
You can use `rownames(installed.packages(priority = 'base'))` instead of hardcoding the package names.

Reply:
Aha! I thought there was a better way (but I couldn't find it). Thanks!